v-next/hardhat/src/internal/cli/prompt/prompt.ts

```diff
@@ -0,0 +1,59 @@
+/**
+ * Display a confirmation prompt to the user. The prompt will be canceled if no response is given within a specified time limit.
+ * @param name Used as the key for the answer on the returned values (answers) object.
+ * @param message The message to display when the prompt is rendered in the terminal.
+ * @param timeoutMilliseconds After how much time the prompt will be cancelled if no answer is given.
+ * @returns True or false based on the user input, or undefined if the prompt times out.
+ */
+export async function confirmationPromptWithTimeout(
+  name: string,
+  message: string,
+  timeoutMilliseconds: number = 10_000,
+): Promise<boolean | undefined> {
+  try {
+    // eslint-disable-next-line @typescript-eslint/consistent-type-assertions -- The types in the module "enquirer" are not properly defined
+    const { default: enquirer } = (await import("enquirer")) as any;
+
+    const prompt = new enquirer.prompts.Confirm(
+      createConfirmationPrompt(name, message),
+    );
+
+    // The timeout is a safety measure in case Hardhat is executed in a CI or another non-interactive environment and we do not detect it.
+    // Instead of blocking the process indefinitely, we abort the consent operation after a while.
+    let timeout;
+    const timeoutPromise = new Promise((resolve) => {
+      timeout = setTimeout(resolve, timeoutMilliseconds);
+    });
+
+    const result: boolean | undefined = await Promise.race([
+      prompt.run(),
+      timeoutPromise,
+    ]);
+
+    clearTimeout(timeout);
+
+    if (result === undefined) {
+      await prompt.cancel();
+      return undefined;
+    }
+
+    return result;
+  } catch (e) {
+    if (e === "") {
+      // If the user cancels the prompt, we quit
+      process.exit();
```

Comment:
it is correct to terminate the process?

Reply:
I think we should return undefined here, as it represents the lack of an answer. This should be clear in the function docs.

In general, we should never exit the process from helper functions like this one, as that's unexpected.

Reply:
Done in [this commit](https://github.com/NomicFoundation/hardhat/pull/5503/commits/0ca57efb65ebadaf11d07d755a750f37bc684bf8)